editor/store/effects.js

```diff
@@ -499,4 +501,15 @@ export default {
 	SELECT_BLOCK: removeProvisionalBlock,
 
 	MULTI_SELECT: removeProvisionalBlock,
+
+	REMOVE_BLOCKS( action, { getState, dispatch } ) {
+		const state = getState();
+		const previousState = { ...state, editor: { present: last( state.editor.past ) } };
+		const firstRemovedBlockUID = action.uids[ 0 ];
+		const rootUID = getBlockRootUID( previousState, firstRemovedBlockUID );
+		const blockUIDToSelect = getPreviousBlockUid( previousState, firstRemovedBlockUID ) || rootUID;
+		if ( blockUIDToSelect ) {
```

Comment:
Would we want to check that the block UID to select is not the currently selected block? I'd been a bit worried about the backspace behavior of removing a block not setting the text caret to the correct position. It seems to work fine, probably because `selectBlock` does nothing if it's set to what's already selected, but seems we could avoid some redundancy here.

Reply:
I added a verification.